test/project_types/script/layers/application/connect_app_test.rb

```diff
@@ -0,0 +1,130 @@
+# frozen_string_literal: true
+
+require "project_types/script/test_helper"
+
+describe Script::Layers::Application::ConnectApp do
+  let(:context) { TestHelpers::FakeContext.new(root: Dir.mktmpdir) }
+  let(:script_name) { "script_name" }
+  let(:language) { "typescript" }
+  let(:extension_point_type) { "payment-methods" }
+
+  let(:script_project_repository) { TestHelpers::FakeScriptProjectRepository.new }
+
+  before do
+    Script::Layers::Infrastructure::ScriptProjectRepository.stubs(:new).returns(script_project_repository)
+    Script::Layers::Infrastructure::ScriptProjectRepository.any_instance.stubs(:get).returns(script_project)
+  end
+
+  describe ".call" do
+    subject do
+      Script::Layers::Application::ConnectApp.call(ctx: context)
+    end
+
+    describe "when script project env is valid" do
+      let(:script_project) do
+        script_project_repository.create(
+          language: language,
+          extension_point_type: extension_point_type,
+          script_name: script_name,
+          env: ShopifyCLI::Resources::EnvFile.new(
```

Comment:
Same comment as above: hopefully, we are not writing anything during unit tests.

Reply:
Gonna post the snippet of our slack convo to explain why I'm leaving this here for this PR.

if we pass in a mock env, it fails because it creates a real `Domain::ScriptProject`, which has smart properties and expects an actual env, via https://github.com/Shopify/shopify-cli/blob/bee47910bc1987a0bad13fdbc74e6cdc5b025953/lib/project_types/script/layers/domain/script_project.rb#L12